### PR TITLE
now ajv validating for either Receipt Object or ArchiverReceipt object

### PR DESCRIPTION
**Pull request by devendra-shardeum**
smoke testing done on the pr.. and testing also done with adding the archiver to existing archiver in devnetUs.. it was able to sync data and couldn't reproduce the issue anymore ... detailed result is shared in the ticket itself : https://linear.app/shm/issue/SHARD-1200/ajv-validation-error-on-archiver-can-cause-missing-receipts#comment-5f4841b3

### CODE REVIEW

**Review by PudgyPug — APPROVED** [review]

**Review by abdulazeem-tk4vr — APPROVED** [review]

File: src/Data/Collector.ts [hunk 1]
```diff
@@ -804,7 +817,7 @@ export const verifyArchiverReceipt = async (
 }
 
 export const storeReceiptData = async (
-  receipts: Receipt.ArchiverReceipt[],
+  receipts: Receipt.Receipt[] | Receipt.ArchiverReceipt[],
```
Comment (kun6fup4nd4) [inline src/Data/Collector.ts:820]: This is from archive-server repo: 
export interface Receipt extends ArchiverReceipt
why you need | here ?

Comment (devendra-shardeum) [reply src/Data/Collector.ts:820]: details are here in the ticket : https://linear.app/shm/issue/SHARD-1200/ajv-validation-error-on-archiver-can-cause-missing-receipts#comment-5f4841b3
tl;dr :
1. when archiver syncs data from validator .. receiving object comes of type ArchiverReceipt
2. when archiver syncs data from another archiver .. receiving objects comes of type Receipt

**Review by kun6fup4nd4 — APPROVED** [review]

**Review by aniketdivekar — APPROVED** [review]
